File: web/src/main/java/by/sorface/passport/web/services/users/social/GitHubOAuth2UserService.java
package by.sorface.passport.web.services.users.social;

import by.sorface.passport.web.dao.models.enums.ProviderType;
import by.sorface.passport.web.records.socialusers.GitHubOAuth2User;
import by.sorface.passport.web.services.users.RoleService;
import by.sorface.passport.web.services.users.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class GitHubOAuth2UserService extends AbstractSocialOAuth2UserService<GitHubOAuth2User>
        implements SocialOAuth2UserService<GitHubOAuth2User> {

    @Autowired
    public GitHubOAuth2UserService(final UserService userService,
                                   final RoleService roleService) {
        super(ProviderType.GITHUB, userService, roleService);
    }

}

File: backend/src/main/kotlin/by/sorface/idp/service/oauth/jdbc/GitHubOAuth2UserService.kt
package by.sorface.idp.service.oauth.jdbc

import by.sorface.idp.config.security.oauth2.records.GitHubOAuth2User
import by.sorface.idp.dao.sql.model.enums.ProviderType
import by.sorface.idp.dao.sql.repository.user.RoleRepository
import by.sorface.idp.dao.sql.repository.user.UserRepository
import by.sorface.idp.service.oauth.SocialOAuth2UserService
import org.springframework.stereotype.Service

@Service
class GitHubOAuth2UserService(
    userRepository: UserRepository,
    roleRepository: RoleRepository
) : AbstractSocialOAuth2UserService<GitHubOAuth2User>(ProviderType.GITHUB, userRepository, roleRepository), SocialOAuth2UserService<GitHubOAuth2User>